node.go

```diff
@@ -21,6 +21,9 @@ import (
 // Nodes are immutable values and therefore safe to use concurrently from
 // multiple goroutines.
 type Node interface {
+	// Returns field id
```

Comment:
Let's document this a bit more:
- explain that this is intended to be the protobuf field id in the parquet specification
- state that it is only useful on leaf nodes
- document the valid range and zero means "no field id"

See for example the documentation on other methods of the `Node` type, we try to clearly describe anything that isn't trivial.

Reply:
done 480e151

I tried my best, we can do better though.

> explain that this is intended to be the protobuf field id in the parquet specification

No its not! This is parquet `field_id` from specification. This PR has nothing to do with protobuf its about `field_id` from the specification.

> state that it is only useful on leaf nodes

Not true, this is for all valid nodes, ie leafs and non leafs.  I updated tests to include other nodes. And  the spec doesn't define how this field is used, we don't have to define how we use it for now.